tests/Type/ValidationTest.php

```diff
@@ -1346,28 +1328,6 @@ public function testRejectsWithReleventLocationsForANonOutputTypeAsAnObjectField
         );
     }
 
-    // DESCRIBE: Type System: Interface fields must have output types
-
-    /**
-     * @see it('rejects an Object implementing a non-type values')
-     */
-    public function testRejectsAnObjectImplementingANonTypeValues() : void
```

Comment:
Removed this because now that `ObjectType::getInterfaces` calls `Schema::resolveType`, anything that is not a `Type` will violate the type hint and throw a natural error.

Reply:
@vladar I was thinking; if you really want this `validate` stuff to work the way it used to, maybe what I could do is change the type hint to an `assert`, and I could throw a crafted exception. I could then catch that in the validation code, and create an `Error` as before. Let me know what you think.

Reply:
Yeah, I am curious which error message and line/string will users get without this? Because right now it should be obvious what is wrong with typedef. But if you get some cryptic error message in an unrelated place of code - the DX is very different

Reply:
Fair enough. I might start another PR to try to advance some of this assert stuff, which should smooth this out and make phpstan level 4 easier to do.

Reply:
I think this is fine to leave out for now. By using stacktraces, users should still be able to figure out the error just fine. `assert()` will improve this, too :+1: